iam-common/src/main/java/it/infn/mw/iam/service/aup/DefaultAupSignatureCheckService.java

```diff
@@ -53,7 +53,7 @@ public boolean needsAupSignature(IamAccount account) {
 
     Date now = new Date(timeProvider.currentTimeMillis());
 
-    if (!aup.isPresent()) {
+    if (!aup.isPresent() || account.isServiceAccount()) {
```

Comment:
I'd separate the two conditions to have two different log messages. In case of service accounts a message like this "_AUP signature not needed for account '{}': Account is a service account_".

Reply:
New if condition added